miden-lib/build.rs

```diff
@@ -20,7 +20,7 @@ use walkdir::WalkDir;
 // ================================================================================================
 
 /// Defines whether the build script can write to /src.
-const CAN_WRITE_TO_SRC: bool = option_env!("DOCS_RS").is_none();
+const BUILD_GENERATED_FILES: bool = option_env!("BUILD_GENERATED_FILES").is_some();
```

Comment:
Maybe we can add to this doc:

```rust
/// Defines whether the build script should generate files in `/src`.
/// The docs.rs build pipeline has a read-only filesystem, so we have to avoid writing to `src`,
/// otherwise the docs will fail to build there. Note that writing to `OUT_DIR` is fine.
```